src/github.com/getlantern/flashlight/withclient/withclient.go

```diff
@@ -0,0 +1,138 @@
+package withclient
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/getlantern/golog"
+
+	"github.com/getlantern/flashlight/client"
+	"github.com/getlantern/flashlight/config"
+	"github.com/getlantern/flashlight/util"
+)
+
+var (
+	log = golog.LoggerFor("withclient")
+)
+
+// The following are mostly meant to
+//   (i)   keep track of directly fronted Dialers, closing them when they
+//         get superseded by a config update.
+//   (ii)  avoid creating more Dialers than needed (one per config update).
+//   (iii) make these dialers available to the config custom poll, without
+//         introducing globals.
+//   (iv)  present a uniform interface in the server side, currently for
+//         the benefit of the config custom poll.
+//
+// Thus, most of the functionality is meant for the client side, and we
+// use mostly placeholders in the server side.  Because of this and for
+// agility of exposition, the comments below refer to the client side.
+
+type clientWithCloseThunk struct {
+	// A particular reference to the directly fronted http.Client (aka fronter).
+	client *http.Client
+	// Placeholder for any cleanup required when this reference is no longer
+	// needed.  We keep track of currently used references with a WaitGroup,
+	// and we use this function to signal we're Done() with this one.
+	close func()
+}
+
+type clientMaker struct {
+	// Generates structures like the one above.  A closure over a fronted.Dialer.
+	make func() clientWithCloseThunk
```

Comment:
Maybe a name other than make just to avoid confusion with the make builtin function?